osu.Game.Rulesets.Osu/Mods/OsuModRandom.cs

```diff
@@ -37,23 +35,51 @@ public void ApplyToBeatmap(IBeatmap beatmap)
 
             var positionInfos = OsuHitObjectGenerationUtils.GeneratePositionInfos(osuBeatmap.HitObjects);
 
-            float rateOfChangeMultiplier = 0;
+            float sequenceOffset = 0;
+            bool flowDirection = false;
 
-            foreach (var positionInfo in positionInfos)
+            for (int i = 0; i < positionInfos.Count; i++)
             {
-                // rateOfChangeMultiplier only changes every 5 iterations in a combo
-                // to prevent shaky-line-shaped streams
-                if (positionInfo.HitObject.IndexInCurrentCombo % 5 == 0)
-                    rateOfChangeMultiplier = (float)rng.NextDouble() * 2 - 1;
+                bool invertFlow = false;
+
+                if (i == 0 ||
+                    (positionInfos[i - 1].HitObject.NewCombo && (i <= 1 || !positionInfos[i - 2].HitObject.NewCombo) && (i <= 2 || !positionInfos[i - 3].HitObject.NewCombo)) ||
```

Comment:
This condition needs to be factored into helper local methods more. I could barely tell what this was doing (namely, checking if any of the three objects in the sequence start a new combo).

Reply:
Instead of creating a separate method, I used `HitObject.IndexInCurrentCombo` to make it easier to understand. Let me know if you think that it's still to complex.

Reply:
Nevermind, I did end up putting them into their own methods (https://github.com/ppy/osu/pull/18769/commits/8c624d3269902ea8f83e874fbfebd204c97b8206).